### PR TITLE
Code improvement EncryptionResolverProvider and configuration settings validation

### PR DESCRIPTION
**Pull request by fennekit**
## Description

Cleanup of Encryption resolver. Reduces code complexity in this class. Add setting validation

## Quality checklist

- [x] Your code complies with our [Coding Style](https://github.com/SteeltoeOSS/.github/blob/main/contributing-docs/contributing-code-style.md).
- [x] You've updated unit and/or integration tests for your change, where applicable.
- [x] You've updated documentation for your change, where applicable.
      If your change affects other repositories, such as [Documentation](https://github.com/SteeltoeOSS/Documentation), [Samples](https://github.com/SteeltoeOSS/Samples) and/or [MainSite](https://github.com/SteeltoeOSS/MainSite), add linked PRs here.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.
- [x] You've added required license files and/or file headers (explaining where the code came from with proper attribution), where code is copied from StackOverflow, a blog, or OSS.

### CODE REVIEW

File: src/Configuration/src/Encryption/Decryption/EncryptionFactory.cs [hunk 1]
```diff
@@ -25,4 +27,18 @@ public static ITextDecryptor CreateEncryptor(ConfigServerEncryptionSettings sett
...
     }
+
+    private static void EnsureValidEncryptionSettings(ConfigServerEncryptionSettings settings)
+    {
+        bool validEncryptionKeySettings = !string.IsNullOrEmpty(settings.EncryptionKey);
+
+        bool validKeystoreSettings = !string.IsNullOrEmpty(settings.EncryptionKeyStoreLocation) && !string.IsNullOrEmpty(settings.EncryptionKeyStorePassword) &&
+            !string.IsNullOrEmpty(settings.EncryptionKeyStoreAlias);
+
+        if (!(validEncryptionKeySettings ^ validKeystoreSettings))
```
Comment (bart-vmware) [inline src/Configuration/src/Encryption/Decryption/EncryptionFactory.cs:38]:
```suggestion
        if (!validEncryptionKeySettings || !validKeystoreSettings)
```

Comment (hananiel) [inline src/Configuration/src/Encryption/Decryption/EncryptionFactory.cs:38]:
```suggestion
        if (validEncryptionKeySettings && validKeystoreSettings)
```
This reads slightly clearer to me , but feel free to leave as is if you feel strongly about it.

**Review by hananiel — APPROVED** [review]
See the comment and resolve as you see fit. If changes require re-approval, I will be happy to.

File: src/Configuration/src/Encryption/Decryption/EncryptionFactory.cs (hunk 1 above)
Comment (bart-vmware) [reply src/Configuration/src/Encryption/Decryption/EncryptionFactory.cs:38]: Funny, I made a similar comment yesterday. Then I read the error message, realized the XOR was intentional, and deleted my comment. The code is correct and concise. Yet somehow it triggered both of us. Personally, I'd rewrite this to `&&`/`||` logical operators to improve code readability. Exclusive ORs (and to a lesser degree, bitwise operators in general) are more esoteric. I'm fine with keeping it this way, though.

**Review by bart-vmware — APPROVED** [review]

**Comment by bart-vmware** [comment]
@fennekit When I deploy an application, it shows a warning at startup. Is that intentional?
```
2023-08-09T15:53:12.978927182+02:00 Setting ASPNETCORE_URLS=http://0.0.0.0:8080
2023-08-09T15:53:13.578498289+02:00 info: Steeltoe.Management.Diagnostics.DiagnosticObserver[0]
2023-08-09T15:53:13.578521789+02:00       DiagnosticObserver TraceDiagnosticObserver Subscribed to Microsoft.AspNetCore
2023-08-09T15:53:13.578532089+02:00 info: Steeltoe.Management.Diagnostics.DiagnosticObserver[0]
2023-08-09T15:53:13.578535489+02:00       DiagnosticObserver AspNetCoreHostingObserver Subscribed to Microsoft.AspNetCore
2023-08-09T15:53:13.593481794+02:00 warn: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager[35]
2023-08-09T15:53:13.593501794+02:00       No XML encryptor configured. Key {83a14665-adaa-4bd6-b238-b93853255f32} may be persisted to storage in unencrypted form.
2023-08-09T15:53:13.669726629+02:00 info: Microsoft.Hosting.Lifetime[14]
2023-08-09T15:53:13.669753829+02:00       Now listening on: http://0.0.0.0:8080
2023-08-09T15:53:13.670411334+02:00 info: Microsoft.Hosting.Lifetime[0]
2023-08-09T15:53:13.670425434+02:00       Application started. Press Ctrl+C to shut down.
2023-08-09T15:53:13.670430034+02:00 info: Microsoft.Hosting.Lifetime[0]
2023-08-09T15:53:13.670434534+02:00       Hosting environment: Production
2023-08-09T15:53:13.670438134+02:00 info: Microsoft.Hosting.Lifetime[0]
2023-08-09T15:53:13.670442034+02:00       Content root path: /workspace/
```

**Pull request merged**